virtual_ecosystem/models/abiotic/abiotic_tools.py

```diff
@@ -122,3 +127,115 @@ def find_last_valid_row(array: NDArray[np.float32]) -> NDArray[np.float32]:
             new_row.append(np.nan)
 
     return np.array(new_row)
+
+
+def calculate_dewpoint_temperature(
+    air_temperature: NDArray[np.float32],
+    effective_vapour_pressure_air: NDArray[np.float32],
+) -> NDArray[np.float32]:
+    """Calculate the dewpoint temperature.
+
+    Args:
+        air_temperature: Air temperature, [C]
+        effective_vapour_pressure_air: Actual vapour pressure, [kPa]
+
+    Returns:
+        Dewpoint temperature, [C]
+    """
+
+    e0_positive = 611.2 / 1000  # e0 in kPa
+    latent_heat_vapourisation = (2.501 * 10**6) - (2340 * air_temperature)
+    dewpoint_temperature_positive = 1 / 273.15 - (
+        461.5 / latent_heat_vapourisation
+    ) * np.log(effective_vapour_pressure_air / e0_positive)
+
+    e0_negative = 610.78 / 1000
+    latent_heat_sublimation = 2.834 * 10**6
+    dewpoint_temperature_negative = 1 / 273.16 - (
+        461.5 / latent_heat_sublimation
+    ) * np.log(effective_vapour_pressure_air / e0_negative)
+
+    return np.where(
+        air_temperature >= 0,
+        1 / dewpoint_temperature_positive - 273.15,
+        1 / dewpoint_temperature_negative - 273.15,
+    )
+
+
+def calculate_slope_of_saturated_pressure_curve(
+    temperature: NDArray[np.float32],
+    saturated_pressure_slope_parameters: list[float],
+) -> NDArray[np.float32]:
+    r"""Calculate slope of the saturated pressure curve.
+
+    Args:
+        temperature: Temperature, [C]
+        saturated_pressure_slope_parameters: List of parameters to calculate
+            the slope of the saturated vapour pressure curve
+
+    Returns:
+        Slope of the saturated pressure curve, :math:`\Delta_{v}`
+    """
+
+    return (
+        saturated_pressure_slope_parameters[0]
```

Comment:
Are these parameters basically just coefficients to a polynomial (I've never been clear how to handle/name these)?

Reply:
Yes, they are. A better way of dealing with this is probably with `gettattr()` like this (from abiotic_simple):
```lower, upper = getattr(bounds, "soil_temperature")```